jaxrs-api/src/main/java/javax/ws/rs/client/ClientBuilder.java

```diff
@@ -72,30 +72,29 @@ public static ClientBuilder newBuilder() {
                 Class pClass = ClientBuilder.class;
                 String classnameAsResource = pClass.getName().replace('.', '/') + ".class";
                 final SecurityManager sm = System.getSecurityManager();
-                ClassLoader loader = null;
+                final String errorMessage;
                 if (sm == null) {
-
-                    loader = pClass.getClassLoader();
+                    ClassLoader loader = pClass.getClassLoader();
                     if (loader == null) {
                         loader = ClassLoader.getSystemClassLoader();
                     }
-
+                    URL targetTypeURL = loader.getResource(classnameAsResource);
+                    errorMessage = "ClassCastException: attempting to cast"
+                            + delegate.getClass().getClassLoader().getResource(classnameAsResource)
+                            + " to " + targetTypeURL;
```

Comment:
These lines and lines 92-94 are duplicates.  Can't this be consolidated into a single point of definition?

Reply:
I thought about it. Really it's just the `delegate.getClass().getClassLoader()` that needs to be privileged, but there didn't seem to be a good way to break that up without two privileged blocks so I did it like this.